daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -37,6 +37,49 @@ import scala.jdk.CollectionConverters._
 import scala.collection.immutable.TreeSet
 import scala.math.Ordering.Implicits.infixOrderingOps
 
+private[speedy] abstract class SpeedyBuiltin(val arity: Int) {
+
+  private[speedy] def execute(args: util.ArrayList[SValue], machine: Machine): Control
+
+  protected def crash(msg: String): Nothing =
+    throw SErrorCrash(getClass.getCanonicalName, msg)
+
+  protected def unexpectedType(i: Int, expected: String, found: SValue): Nothing =
+    crash(s"type mismatch of argument $i: expected $expected but got $found")
+}
+
+/** Extensions of this mixin trait ensure that builtin argument lists have a length matching the builtin's arity.
+  */
+private[speedy] trait BuiltinArityChecking { inner: SpeedyBuiltin =>
+
+  @inline
+  private[speedy] def execute(args: util.ArrayList[SValue], machine: Machine): Control = {
```

Comment:
These mixins don't work! -- The `execute` methods of both `BuiltinArityChecking` and `BuiltinTokenChecking` are never called.